samples/nrf9160/nrf_cloud_mqtt_multi_service/src/cloud_connection.c

```diff
@@ -19,173 +19,69 @@
-{
-	return k_event_wait(&cloud_connection_events, CLOUD_DISCONNECTED, false, timeout) != 0;
+	return k_event_wait(&cloud_events, CLOUD_DISCONNECTED, false, timeout) != 0;
 }
 
 bool await_date_time_known(k_timeout_t timeout)
 {
-	return k_event_wait(&cloud_connection_events, DATE_TIME_KNOWN, false, timeout) != 0;
+	return k_event_wait(&cloud_events, DATE_TIME_KNOWN, false, timeout) != 0;
 }
 
-/**
- * @brief Notify that network connection has been established.
+/* Delayable work item for handling cloud readiness timeout.
+ * The work item is scheduled at a delay any time connection starts and is cancelled when cloud
+ * readiness is achieved. If the work item executes, that means readiness was not achieved within
+ * the delay, and the connection should be reset.
  */
-static void notify_network_connected(void)
+static void ready_timeout_work_fn(struct k_work *work)
 {
-	k_event_post(&network_connection_events, NETWORK_CONNECTED);
+	ARG_UNUSED(work);
+	LOG_INF("nRF Cloud did not become ready in time, disconnecting and retrying...");
+	disconnect_cloud();
 }
 
-/**
- * @brief Notify that cloud connection is ready.
- */
-static void notify_cloud_ready(void)
-{
-	k_event_post(&cloud_connection_events, CLOUD_READY);
-}
-
-/**
- * @brief Notify that connection to nRF Cloud has been established.
- */
-static void notify_cloud_connected(void)
-{
-	k_event_post(&cloud_connection_events, CLOUD_CONNECTED);
-}
-
-/**
- * @brief Notify that the current date and time have become known.
- */
-static void notify_date_time_known(void)
-{
-	k_event_post(&cloud_connection_events, DATE_TIME_KNOWN);
-}
-
-/**
- * @brief Notify that a cloud association request has been made.
- */
-static void notify_cloud_requested_association(void)
-{
-	k_event_post(&cloud_connection_events, CLOUD_ASSOCIATION_REQUEST);
-}
-
-/**
- * @brief Reset the network connection event flag.
- */
-static void clear_network_connected(void)
-{
-	k_event_set(&network_connection_events, 0);
-}
-
-/**
- * @brief Clear nRF Cloud connection events.
- */
-static void clear_cloud_connection_events(void)
-{
-	k_event_set(&cloud_connection_events, 0);
-}
-
-/**
- * @brief Check whether a user association request has been received from nRF Cloud.
- *
- * If true, we have received an association request, and we must restart the nRF Cloud connection
- * after association succeeds.
- *
- * This flag is reset by the reconnection attempt.
- *
- * @return bool - Whether we have been requested to associate with an nRF Cloud user.
- */
-static bool cloud_has_requested_association(void)
-{
-	return k_event_wait(&cloud_connection_events, CLOUD_ASSOCIATION_REQUEST,
-			    false, K_NO_WAIT) != 0;
-}
+static K_WORK_DELAYABLE_DEFINE(ready_timeout_work, ready_timeout_work_fn);
 
-bool cloud_is_connected(void)
+static void start_readiness_timeout(void)
 {
-	return k_event_wait(&cloud_connection_events, CLOUD_CONNECTED, false, K_NO_WAIT) != 0;
+	LOG_DBG("Starting cloud readiness timeout for %d seconds",
```

Comment:
isn't "the cloud" always ready?  (unless of course AWS is down 🙈 )

Reply:
I think I've made it clearer, let me know if it is sufficient now

Reply:
you have in your above comment: `when the connection to nRF Cloud becomes ready to use`...
this is what we are waiting on, the device+cloud handshaking to be complete.

the log message on line 69 again makes it sound like the cloud is unresponsive: `nRF Cloud did not become ready in time`... which could be the case, but rather uncommon.

i think "connection ready" is more accurate than "cloud ready".
not a big deal, it is up to you.

Reply:
You're right, I'll adjust that phrasing